src/admin/hasura/hasura.service.ts

```diff
@@ -0,0 +1,66 @@
+import { Injectable, InternalServerErrorException, Logger, NotImplementedException } from '@nestjs/common';
+import { ConfigResolverService } from '../../api/config.resolver.service';
+import { UUID } from '@fusionauth/typescript-client';
+import { HttpService } from '@nestjs/axios';
+import { lastValueFrom, map } from 'rxjs';
+
+@Injectable()
+export class HasuraService {
+  private readonly hasuraGraphqlUrl: string;
+  private readonly hasuraGraphqlSecret: string;
+  private readonly mutations: object;
+
+  protected readonly logger = new Logger(HasuraService.name); // logger instance
+
+  constructor(
+    applicationId: UUID,
+    private readonly configResolverService: ConfigResolverService,
+    private readonly httpService: HttpService,
+  ) {
+    const hasuraConfig = configResolverService.getHasura(applicationId);
+    if (!hasuraConfig) {
+      throw new NotImplementedException(`Hasura config not supported/implemented for application: ${applicationId}.`);
+    }
+    this.hasuraGraphqlUrl = hasuraConfig.graphql_url;
+    this.hasuraGraphqlSecret = hasuraConfig.admin_secret;
+    this.mutations = hasuraConfig.mutations
+  }
+
+  async hasuraGraphQLCall(
+    mutationKey: string,
+    payload: object,
+    url: string = this.hasuraGraphqlUrl,
```

Comment:
Keep URL outside of the function params.

Reply:
Not applicable as of now; because the URL is not global. It is defined per `applicationId` in the generic config.